pkg/extender/extender.go

```diff
@@ -847,3 +928,172 @@ func (e *Extender) processCSIExtPodPrioritizeRequest(
+		return fmt.Errorf("unable to inspect PVC: %v err: %w", vmi.RootDiskPVC, err)
+	}
+	storklog.PodLog(pod).Debugf("PV for scoring: %v", pvName)
+
+	volInfo, err := e.Driver.InspectVolume(pvName)
+	if err != nil {
+		return fmt.Errorf("unable to inspect PV: %v. err: %w", pvName, err)
+	}
+
+	if podBeingLiveMigrated != nil {
+		// We give lower score to replica nodes so that if a pod is unable to schedule on the node
+		// where the volume is attached we can run it on a non replica node, so that a subsequent
+		// LiveMigration can move the pod back to the replica node in a single hop.
+		podIsRunningOnVolAttachedNode := e.isPodRunningOnVolAttachedNode(podBeingLiveMigrated, volInfo)
+
+		if podIsRunningOnVolAttachedNode {
+			// LiveMigration in progress and existing pod is running on node with volume attached => Antihyperconvergence
+			// i.e. Give lower scores to nodes with volume replicas
+			storklog.PodLog(pod).Infof("Pod %v is running with same VMI owner reference is using localAttachment.", podBeingLiveMigrated.Name)
+			e.updateVirtLauncherPodPrioritizeScores(encoder,
+				args,
+				volInfo,
+				false, /*preferLocalAttachment*/
+				true /*antihyperconvergence*/)
+		} else {
+			// LiveMigration in progress but existing pod is using NFS.
+			// Use Antihyperconvergence to give lower score to replica nodes without volume attachment
+			// Use preferLocalAttachment to give highest score to node with volume attachment
+
+			e.updateVirtLauncherPodPrioritizeScores(encoder,
+				args,
+				volInfo,
+				true, /*preferLocalAttachment*/
+				true /*antihyperconvergence*/)
+		}
+	} else {
+		// Default behavior for LiveMigration not in progress is hyperconvergence
+		// We should still give a higher score to node with volume attachment. We might not
+		// have attachment info avilable in this scenario and the same score will be
+		// applied for all the replica nodes. However, if volume attachment info is available
+		// that node should get higher score. All non replica nodes will get lower score.
+		e.updateVirtLauncherPodPrioritizeScores(encoder,
+			args,
+			volInfo,
+			true, /*preferLocalAttachment*/
+			false /*antihyperconvergence*/)
+	}
+	return nil
+}
+
+func (e *Extender) updateVirtLauncherPodPrioritizeScores(
+	encoder *json.Encoder,
+	args schedulerapi.ExtenderArgs,
+	volInfo *volume.Info,
+	preferLocalAttachment bool,
+	antihyperconvergence bool) {
+	pod := args.Pod
+
+	// Default behavior is no special preference to local attachment
+	localNodeScore := int64(nodePriorityScore)
+	// Default behavior is hyperconvergence
+	replicaNodeScore := int64(nodePriorityScore)
+	remoteNodeScore := int64(defaultScore)
+
+	if antihyperconvergence {
+		remoteNodeScore = int64(nodePriorityScore)
+		replicaNodeScore = int64(defaultScore)
+	}
+
+	// preferLocalAttachment = true but AttachedOn = "" can occur if there is just one pod
+	// getting scheduled and there is no LiveMigration ongoing
+	if preferLocalAttachment && volInfo.AttachedOn != "" {
```

Comment:
nit: this condition does not seem necessary here since we check the same condition when assigning the localNodeScore. So, we can just set the localNodeScore without checking any conditions?